### PR TITLE
#166214160 Users and admins can see borrowing statistics

### PR DESCRIPTION
**Pull request by Lundii**
#### What does this PR do?
It allows user to see their borrowing statistics
#### Description of Task to be completed?
* added test for the function
* created a route to handle the request

#### How should this be manually tested?
* pull or fetch this branch
* run `npm install` to get the necessary packages
* navigate to `GET api/v1/user/lendinghistory` as a verified and active user
* also, navigate to `GET api/v1/admin/lendinghistory` as a verified admin or the superadmin

#### Any background context you want to provide?
No
#### What are the relevant pivotal tracker stories?
[#166214160](https://www.pivotaltracker.com/story/show/166214160)
#### Screenshots (if appropriate)
Screenshot (83)
Screenshot (84)

#### Questions:
No

### CODE REVIEW

**Review by Adekoreday — CHANGES_REQUESTED** [review]
in your test you only need to login once and start using such login token and details , logining in several time will slow down the test overall test and may in the long run leading to us extending test timeouts

**Comment by Lundii** [comment]
> in your test you only need to login once and start using such login token and details , logining in several time will slow down the test overall test and may in the long run leading to us extending test timeouts

If you look at the tests carefully, you will observe that I'm logging in different people and they are logged in only once.

**Review by Ukhu — COMMENTED** [review]
From your screenshot, the data returned in the borrowing history does not contain the name of the author for each book.

**Comment by Lundii** [comment]
> From your screenshot, the data returned in the borrowing history does not contain the name of the author for each book.

Done

**Review by KvNGCzA — CHANGES_REQUESTED** [review]
This route should be `/user/lendinghistroy`

**Comment by Lundii** [comment]
> This route should be `/user/lendinghistroy`

Done

File: server/controllers/BookController.js [hunk 1]
```diff
@@ -75,6 +77,42 @@ class BookController {
...
+      let filter = { userId: id };
+      if (roleName === 'admin' || roleName === 'superadmin') filter = {};
+      const lendingHistory = await getLendingHistory(response, filter, limit, offset);
+      if (lendingHistory && lendingHistory.length) {
+        const dataValues = JSON.parse(JSON.stringify(lendingHistory));
+        dataValues.forEach((data) => {
+          data.Book.Author = data.Book.BookAuthors[0].Author;
+          delete data.Book.BookAuthors;
+        });
+        return responseMessage(response, 200, { status: 'success', dataValues });
```
Comment (Nennyfills) [inline server/controllers/BookController.js:109]: your response should include count, number of pages and current page

**Review by Nennyfills — CHANGES_REQUESTED** [review]
your pagination responses should include count, current page and the number of pages.

**Comment by Lundii** [comment]
I realized this PR was closed. I may have mistakenly closed it. I presently cannot reopen it. I will open a new PR now.

**Pull request closed without merging**